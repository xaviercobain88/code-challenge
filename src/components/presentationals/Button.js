import React from 'react';
import styled from 'styled-components';
import { Button as Btn } from 'react-native';
import * as colors from '../../constants/colors';

const Container = styled.View`
  border-color: ${props => (props.inverted ? 'white' : colors.secondaryColor)};
  margin-horizontal: 10;
  margin-vertical: 10;
  padding-vertical: 5;
  border-radius: 30;
  border-width: 2;
  width: 120;
`;

const Button = props => (
  <Container {...props}>
    <Btn {...props} />
  </Container>
);

export default Button;
