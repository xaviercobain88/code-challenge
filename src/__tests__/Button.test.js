import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../components/presentationals/Button';
import 'jest-styled-components';

const props = {
  inverted: true,
  onPress: () => ({}),
  title: 'PLAY AGAIN!',
  color: 'white',
};

test('renders correctly', () => {
  const tree = renderer.create(<Button {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
