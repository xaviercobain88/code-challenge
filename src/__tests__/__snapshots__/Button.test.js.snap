// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  color="white"
  inverted={true}
  onPress={[Function]}
  style={
    Array [
      Object {
        "borderColor": "white",
        "borderRadius": 30,
        "borderWidth": 2,
        "marginHorizontal": 10,
        "marginVertical": 10,
        "paddingVertical": 5,
        "width": 120,
      },
      undefined,
    ]
  }
  title="PLAY AGAIN!"
>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={undefined}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "white",
            },
          ]
        }
      >
        PLAY AGAIN!
      </Text>
    </View>
  </View>
</View>
`;
